pkg/storage/tpr/storage_interface.go

```diff
@@ -135,50 +149,60 @@ func (t *store) Create(
 	return nil
 }
 
-// Delete removes the specified key and returns the value that existed at that spot.
-// If key didn't exist, it will return NotFound storage error.
+// Delete fetches the resource at key, removes its finalizer, updates it, and returns the
+// resource before its finalizer was removed.
 //
-// In this implementation, Delete will not write the deleted object back to out
+// If key didn't exist, it will return NotFound storage error.
 func (t *store) Delete(
 	ctx context.Context,
 	key string,
 	out runtime.Object,
 	preconditions *storage.Preconditions,
 ) error {
+	// create adds the get the object remove its finalizer, and
 	ns, name, err := t.decodeKey(key)
 	if err != nil {
 		glog.Errorf("decoding key %s (%s)", key, err)
 		return err
 	}
-
-	req := t.cl.Delete().AbsPath(
-		"apis",
-		groupName,
-		tprVersion,
-		"namespaces",
+	if t.hardDelete {
```

Comment:
the names imply `GracefulDelete` would be the inverse of `hardDelete`, but I can't speak specifics because I have not seen us use `GracefulDelete` in our API server, nor have I seen `GracefulDelete` in the registry